src/components/Choropleth.vue

```diff
@@ -86,18 +89,19 @@ const { map, dataSummary, lockBounds, updateLeafletMap, handleMapBoundsUpdated,
     layerOnEvents
 );
 useLoadingSpinner(map, mapLoading);
+const { selectedFeatures } = useSelectedMapInfo();
```

Comment:
`selectedFeatures` and `selectedIndicators` were getters before so why did I extract them out into a composable?? great question, the watcher was firing as soon as the `mapSetting` changed, **before** the getters for the `selectedFeatures` had fired so the `updateMap` function was getting old `selectedFeatures`, how annoying, but as composables they are fine, getters are probably near the end in the chain of side effects that vue decides to resolve and watchers are probably near the first.

Reply:
i think you're right about the computed for each instance of composable use, i couldnt find anything online about vue optimisation, i think the performance is alright but we should probably address this so it doesn't scale, i think there is a very easy way to address this related to your comment about composables not using other composables, everything that uses these computed is in the `Choropleth` so ill just pass these refs into these other composables! should fix performance concerns

Reply:
update on this, have only kept the composable in choropleth, legend vue files and useDataSummary composable, passed it in everywhere else, should scale a bit better

Reply:
Nice one :+1: